### ISSUE TITLE
in the carbon example, /node_modules/next/dist needs to be explicitly included

### ISSUE THREAD

**Issue by yuuuxt** [docker-slim/docker-slim#211]
Expected Behavior
=================

the [carbon example](https://github.com/docker-slim/examples/tree/master/3rdparty/carbon-now-sh) isn't working as expected.

according to [this](https://github.com/docker-slim/docker-slim/issues/150#issuecomment-650852974) comment, I thought I don't need to explicitly include (part of) `node_modules`.

---
Actual Behavior
=================

I need to have `/node_modules/next/dist` to be explicitly included.

---
Steps to Reproduce the Problem
===============================

  1. docker pull yuuuxt/carbon:4.8.1
  2. (not working) `docker-slim build --copy-meta-artifacts . --include-path=/app/carbon/.next --include-path=/app/carbon/public yuuuxt/carbon:4.8.1`
  3. (working) `docker-slim build --copy-meta-artifacts . --include-path=/app/carbon/.next --include-path=/app/carbon/public --include-path=/app/carbon/node_modules/next/dist yuuuxt/carbon:4.8.1`

---
Specifications
=================
  - Version: 1.36.1
  - Platform: WSL2

**Comment by kcq** [comment]
Thanks for opening the issue @yuuuxt ! Curious if you customized your carbon app image (`yuuuxt/carbon:4.8.1`) in anyway or did you just build it with the Dockerfile from the carbon repo?

**Comment by yuuuxt** [comment]
Hi @kcq , thanks for the project and your reply! The `Dockerfile` is [here](https://github.com/yuuuxt/carbon-slim/blob/master/4.8.1/Dockerfile), I just used `git clone` instead of copying from context, I think they are alike?

**Comment by rqi14** [comment]
I have a similar issue slimming perplexica docker images

**Comment by kcq** [comment]
@rqi14 just double checking... this project, right: https://github.com/ItzCrazyKns/Perplexica/tree/master

**Comment by rqi14** [comment]
> @rqi14 just double checking... this project, right: https://github.com/ItzCrazyKns/Perplexica/tree/master

Yes. I have a demo of building and slimming images here https://github.com/rqi14/Perplexica-docker

I have to specify several include paths

**Comment by ls-ramos** [comment]
Yeah I had similar issue, I was testing to slim the docker image of a simple frontend project using nextJS (v15) (basically just a fresh example project created with `npx create-next-app@latest`). 

I had to add at least `--include-path=/app/node_modules/styled-jsx --include-path=/app/node_modules/next/dist` to the slim build command to get the image running properlly

**Comment by kcq** [comment]
> Yeah I had a similar issue, I was testing to slim the docker image of a simple frontend project using nextJS (v15) (basically just a fresh example project created with `npx create-next-app@latest`).
> 
> I had to add at least `--include-path=/app/node_modules/styled-jsx --include-path=/app/node_modules/next/dist` to the slim build command to get the image running properlly

I'll try to repro with an `npx create-next-app@latest` example. In the meantime, the `--include-app-next-dir`, `--include-app-next-dist-dir`, `--include-app-next-build-dir`, `--include-app-next-nodemodules-dir` and `--include-app-next-static-dir` flags might be useful too.
